hles/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala

```diff
@@ -36,11 +42,20 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
     * Get the singleton value for an attribute in this record, if one exists.
     *
     * Raises an error if the attribute has multiple values.
+    *
+    * Only unique duplicates will throw an error
+    * If we receive multiple fields with the same value, the head of the set is used
     */
   def getOptional(field: String): Option[String] = {
-    val values = fields.getOrElse(field, Array.empty)
-    if (values.length > 1) {
-      throw new IllegalStateException(s"Record $id has multiple values for field $field")
+    val values = fields.getOrElse(field, Array.empty).toSet
+    // If there are multiple values
+    if (values.size > 1) {
+      // Filtering out "NA"
+      if (values.--(Array("NA")).size > 1) {
+        throw new IllegalStateException(s"Record $id has multiple values for field $field")
+      } else {
+        values.headOption
```

Comment:
are we worried about this possibly returning `NA`?

Reply:
Yes, we were seeing `NA` values being pulled by getOption